src/main/java/xades4j/verification/DistinguishedNameComparer.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * XAdES4j - A Java library for generation and verification of XAdES signatures.
+ * Copyright (C) 2018 Luis Goncalves.
+ *
+ * XAdES4j is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 3 of the License, or any later version.
+ *
+ * XAdES4j is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License along
+ * with XAdES4j. If not, see <http://www.gnu.org/licenses/>.
+ */
+package xades4j.verification;
+
+import com.google.inject.Inject;
+import javax.security.auth.x500.X500Principal;
+import org.bouncycastle.asn1.x500.X500Name;
+import xades4j.providers.X500NameStyleProvider;
+
+/**
+ * <b>Experimental API</b>. It may be changed or removed in future releases.
+ *
+ * @author luis
+ */
+class DistinguishedNameComparer
+{
+    private final X500NameStyleProvider x500NameStyleProvider;
+    
+    @Inject
+    DistinguishedNameComparer(X500NameStyleProvider x500NameStyleProvider)
+    {
+        this.x500NameStyleProvider = x500NameStyleProvider;
+    }
+
+    /**
+     * @exception IllegalArgumentException if the DN string is invalid 
+     */
+    boolean areEqual(X500Principal parsedDn, String stringDn)
+    {
+        X500Name first = X500Name.getInstance(parsedDn.getEncoded());
+        // TODO consider simplifying this by constructing from string and using the configured keyword map
+        X500Name second = X500Name.getInstance(this.x500NameStyleProvider.fromString(stringDn).getEncoded());
```

Comment:
Since `x500NameStyleProvider.fromString(stringDn)` already considers the "name style", won't the encoded version preserve the proper OIDs?

If using the "name style" is really needed, we could as well use `X500Name#getInstance(org.bouncycastle.asn1.x500.X500NameStyle, string)` directly?
 `

Reply:
Yes, we can use `new X500Name(org.bouncycastle.asn1.x500.X500NameStyle, string)` directly. But in this case we have ambiguously behavior with 2 points `xades4j.providers.impl.DefaultX500NameStyleProvider#DefaultX500NameStyleProvider(xades4j.utils.X500ExtensibleNameStyle)` and `xades4j.verification.DistinguishedNameComparer#DistinguishedNameComparer`. So I suggest to use one point for working with DN.




P.S. Sorry for response delay